### PR TITLE
Replace our mock data with the actual api request

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Issue:** https://github.com/department-of-veterans-affairs/va.gov-team/issues/4147

**Current Behavior:** We use mock search results on staging.va.gov/find-va-forms-beta.

**Desired Behavior:** We use actual search results from vets-api on staging.va.gov/find-va-forms-beta.

This PR implements the above desired behavior.

## Testing done
Locally.

## Screenshots
image

## Acceptance criteria
- [x] Make API request to `[domain]/v0/forms` instead of mock data
- [x] Remove mock data from repo

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/find-va-forms/api/index.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
+export const fetchFormsApi = async query => {
+  let FORMS_URL = '/forms';
 
-  const forms = {
-    data: filteredData,
-  };
+  // Add the `query` query param if provided.
+  if (query) {
+    FORMS_URL = `${FORMS_URL}?query=${query}`;
+  }
```
Comment (ncksllvn) [inline src/applications/find-va-forms/api/index.js]: We have an `append-query` module to help encode this. Not sure what it does under the hood other than just a series of `encodeUriComponent`s though.

File: src/applications/find-va-forms/api/index.js [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+
+  // Make the request for the forms.
+  const response = await apiRequest(FORMS_URL);
+
+  // Derive the forms.
+  const forms = response?.data;
```
Comment (ncksllvn) [inline src/applications/find-va-forms/api/index.js]: I don't think `data` could be missing without an `error` property in its place, but not sure if we want to do something here anyway.

**Review by ncksllvn — APPROVED** [review]
LGTM!

File: src/applications/find-va-forms/api/index.js (hunk 1 above)
Comment (kelsonic) [reply src/applications/find-va-forms/api/index.js]: Updated! 🙂

**Comment by kelsonic** [comment]
This is ready for review! 🎉 @department-of-veterans-affairs/frontend-review-group

**Comment by ncksllvn** [comment]
@kelsonic looks like there's still a unit test to fix -

```
Stacktrace
AssertionError: expected 'findVAForms/FETCH_FORMS_FAILURE' to equal 'findVAForms/FETCH_FORMS_SUCCESS'
    at Context.equal (src/applications/find-va-forms/tests/actions/index.unit.spec.js:97:43)
    at tryCatch (node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:288:22)
    at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:114:21)
    at asyncGeneratorStep (src/applications/find-va-forms/tests/actions/index.unit.spec.js:13:103)
    at _next (src/applications/find-va-forms/tests/actions/index.unit.spec.js:15:194)
```

**Review by rianfowler — APPROVED** [review]

**Comment by kelsonic** [comment]
So it's pretty odd -- if I run the command `yarn test:unit -- src/applications/find-va-forms`, it says all of the unit tests are passing:

image

If I run the command `yarn test:unit`, it fails on the unit test that's using the `apiRequest` module:

image

When I log out why that `try catch` fails, it logs this:

```
TypeError: Cannot read property 'catch' of undefined
    at apiRequest (/Users/kelsonadams/Desktop/business/gcio/dsva/vets-website/src/platform/utilities/api/index.js:82:10)
```

Is it possible that `apiRequest` module is getting modified over the lifespan of the unit tests being run? 😬In theory, it should be consistent across both unit test commands... @rianfowler @ncksllvn

**Comment by ncksllvn** [comment]
@kelsonic very possible! The best way past this is probably to mock `fetchFormsApi` in the actions unit test. I think we can do this by either adding an optional param into `fetchFormsThunk` for `fetchFormsApi` that we can stub from unit tests like we're doing with the `location` parameter already. Or, we can mock `fetch` globally. We have a platform helper, `mockFetch`, at `platform/testing/unit/helpers.js` for doing this.

**Comment by kelsonic** [comment]
Yay it passes! Except for a random e2e test lol

**Pull request merged**